libnmstate/netapplier.py

```diff
@@ -306,6 +310,15 @@ def assert_ifaces_state(ifaces_desired_state, ifaces_current_state):
         iface_cstate = ifaces_current_state[ifname]
         iface_dstate = _canonicalize_desired_state(
             ifaces_desired_state[ifname], iface_cstate)
+
+        iface_ethernet_dstate = iface_dstate.get('ethernet')
```

Comment:
IMHO it should be explained what is going on here to be able to know this still in two weeks. I needed the comment in the sanitize method as well to be able to understand the issue. :-)
Maybe it would be also better to put in its own method to be able to use variables such as current_state and desired_state since `...cstate/dstate` are hard to distinguish.

Reply:
Extracting to smaller methods is always good, thanks for pointing this out.